src/main/java/io/divolte/server/config/MappingConfiguration.java

```diff
@@ -0,0 +1,55 @@
+package io.divolte.server.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableSet;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Objects;
+import java.util.Optional;
+
+@ParametersAreNonnullByDefault
+public class MappingConfiguration {
+    private static final String DEFAULT_DISCARD_CORRUPTED = "false";
+    private static final String DEFAULT_DISCARD_DUPLICATES = "false";
+
+    public final Optional<String> schemaFile;
+    public final Optional<String> mappingScriptFile;
+
+    public final ImmutableSet<String> sources;
+    public final ImmutableSet<String> sinks;
+
+    public final boolean discardCorrupted;
+    public final boolean discardDuplicates;
+
+    @JsonCreator
+    MappingConfiguration(final Optional<String> schemaFile,
+                         final Optional<String> mappingScriptFile,
+                         @JsonProperty(required = true)
```

Comment:
If the reference.conf has a empty collection for the property (meaning the specific conf would add entries), I think it's not possible to not have the property present. In that case the required=true wouldn't really add anything.

We could think about adding `@NotEmpty` to these fields, meaning the config would not validate with zero of either sources, mappings or sinks.